File: Objective-C Version/AnimatedCurveDemo/AnimatedCurveDemo/CurveView/CurveView.m
//
//  CurveView.m
//  AnimatedCurveDemo
//
//  Created by Kitten Yang on 7/6/15.
//  Copyright (c) 2015 Kitten Yang. All rights reserved.
//

#import "CurveView.h"
#import "CurveLayer.h"


@interface CurveView()

@property (nonatomic,strong)CurveLayer *curveLayer;

@end

@implementation CurveView


+ (Class)layerClass{
    return [CurveLayer class];
}

-(id)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        
        
    }
    return self;
}


-(void)setProgress:(CGFloat)progress{

    self.curveLayer.progress = progress;
    [self.curveLayer setNeedsDisplay];
    
    
}

-(void)willMoveToSuperview:(UIView *)newSuperview{

    self.curveLayer = [CurveLayer layer];
    self.curveLayer.frame = self.bounds;
    self.curveLayer.contentsScale = [UIScreen mainScreen].scale;
    self.curveLayer.progress = 0.0f;
    [self.curveLayer setNeedsDisplay];
    [self.layer addSublayer:self.curveLayer];
    
}




@end

File: Swift Version/AnimatedCurveDemo-Swift/AnimatedCurveDemo-Swift/Classes/CurveView/CurveView.swift
//
//  CurveView.swift
//  AnimatedCurveDemo-Swift
//
//  Created by Kitten Yang on 1/18/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

class CurveView: UIView {
    var progress: CGFloat = 0.0 {
        didSet{
            curveLayer.progress = progress
            curveLayer.setNeedsDisplay()
        }
    }

    private var curveLayer: CurveLayer!
    
    override init(frame: CGRect) {
        super.init(frame:frame)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func willMoveToSuperview(newSuperview: UIView?) {
        super.willMoveToSuperview(newSuperview)
        curveLayer = CurveLayer()
        curveLayer.frame = bounds
        curveLayer.contentsScale = UIScreen.mainScreen().scale
        curveLayer.progress = 0.0
        curveLayer.setNeedsDisplay()
        layer.addSublayer(curveLayer)
    }
    
}
